SillyBusinessInc/SillyBusinessGame
# Feat/#107 melee enemies

## Purpose of this PR:
Reworking the enemies to set a solid base for the new enemy behaviors.
Created a new Melee enemy that is highly customizable for the designers.
Add a knockback effect to attacks so the player gets knocked back on hit.

## How to test:
Place down the new Melee Enemy prefab in any scene. It should work out of the box without needing to pass it any additional references.
Check if it switches between idling and roaming as it's supposed to.
Check if it remains in chasing mode once the player has activate it's chasing mode by getting close enough.
Check to see if the enemy does damage, and applies knockback to the player character.


### links: 
closes #107
